### PR TITLE
Add Start.gg provider

### PR DESCRIPTION
**Pull request by nickbeen**
This PR adds an OAuth2 provider for [start.gg](https://www.start.gg) (formerly known as smash.gg), an online provider acquired by Microsoft for hosting tournaments and events for various video games used by individuals as well as companies such as Epic Games, Capcom and Bandai Namco.

Currently start.gg only supports the authorization code grant flow and refresh token.

Docs: https://dev.start.gg/docs/oauth/oauth-overview

### CODE REVIEW

File: src/StartGg/Provider.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function additionalConfigKeys(): array
+    {
+        return [
+            'tournament.manager',
```
Comment (lucasmichot) [inline src/StartGg/Provider.php:109]: Those are unused

File: src/StartGg/Provider.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+/**
+ * @see https://dev.start.gg/docs/oauth/oauth-overview
+ */
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
```
Comment (lucasmichot) [inline src/StartGg/Provider.php:16]: Can you remove this comment?

**Review by lucasmichot — CHANGES_REQUESTED** [review]

File: src/StartGg/Provider.php (hunk 1 above)
Comment (nickbeen) [reply src/StartGg/Provider.php]: I have removed them.

File: src/StartGg/Provider.php (hunk 2 above)
Comment (nickbeen) [reply src/StartGg/Provider.php]: Done. I only added it because several other socialite providers included this comment.

**Review by lucasmichot — APPROVED** [review]
GTM @atymic

**Comment by nickbeen** [comment]
Any ETA for merging this PR? @atymic @lucasmichot

**Review by atymic — APPROVED** [review]

**Pull request merged**